src/modules/45-dashboards/components/AidaGenerating/index.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import { Icon, Text } from '@harness/uicore'
+import { Color, FontVariation } from '@harness/design-system'
+import { useDashboardsContext } from '@dashboards/pages/DashboardsContext'
+import { Message, MessageType, VisualizationType } from '@dashboards/types/AidaTypes.types'
+import { useStrings } from 'framework/strings'
+import { AiAddTileRequestBody, useAiGenerateTile } from 'services/custom-dashboards'
+import css from './AidaGenerating.module.scss'
+
+interface AidaGeneratingProps {
+  messages: Message[]
+}
+
+const getValueFromMessage = (messages: Message[], messageType: MessageType, key?: string): string => {
+  let matchingValue = ''
+
+  messages.find(message => {
+    if (messageType === MessageType.Prompt && message.promptMapping) {
+      const matchingMapping = message.promptMapping.find(mapping => mapping.key === key)
+      if (matchingMapping) {
+        matchingValue = matchingMapping.value
+        return true
+      }
+    }
+    if (messageType === MessageType.Text && message.type === MessageType.Text) {
+      matchingValue = message.content
+      return true
+    }
+  })
+
+  return matchingValue
```

Comment:
This is a really weird way of using a `find`, since you're completely ignoring output. Instead of maintaining a variable and treating it as a `forEach` or even `some`, you can just use the found message.
```suggestion
  const msg = messages.find(message => {
    if (messageType === MessageType.Prompt && message.promptMapping) {
      const matchingMapping = message.promptMapping.find(mapping => mapping.key === key)
      if (matchingMapping) {
        return true
      }
    }
    
    if (messageType === MessageType.Text && message.type === MessageType.Text) {
      return true
    }
  })

  return msg?.content || ''
```

Reply:
This was a late night 🧠💨 . I just needed to use a `for ... of` loop